mailserver/mailserver.go

```diff
@@ -522,29 +523,41 @@ func (s *WMailServer) processRequestInBundles(
 	// Otherwise publish what you have so far, reset the bundle to the
 	// current envelope, and leave if we hit the limit
 	for iter.Next() {
-		var envelope whisper.Envelope
 
-		decodeErr := rlp.DecodeBytes(iter.Value(), &envelope)
-		if decodeErr != nil {
-			log.Error("[mailserver:processRequestInBundles] failed to decode RLP",
-				"err", decodeErr,
-				"requestID", requestID)
-			continue
-		}
+		var envelopeBloom []byte
+		rawValue := make([]byte, len(iter.Value()))
+		copy(rawValue, iter.Value())
+
+		key := iter.Key()
+
+		// Old key, we extract the topic from the envelope
+		if len(key) != DBKeyLength {
+			var envelope whisper.Envelope
+			decodeErr := rlp.DecodeBytes(rawValue, &envelope)
+			if decodeErr != nil {
+				log.Error("[mailserver:processRequestInBundles] failed to decode RLP",
+					"err", decodeErr,
+					"requestID", requestID)
+				continue
+			}
+			envelopeBloom = envelope.Bloom()
 
-		if !whisper.BloomFilterMatch(bloom, envelope.Bloom()) {
+		} else {
+			envelopeBloom = whisper.TopicToBloom(whisper.BytesToTopic(key[timestampLength+common.HashLength:]))
```

Comment:
can we have a `Topic()` method in `DBKey` struct to make this part a bit more readable?